src/intfuncs.c

```diff
@@ -620,6 +620,293 @@ Obj FuncINT_STRING ( Obj self, Obj string )
 
 /****************************************************************************
 **
+*F SmallInt Bitmask operations
+*
+*  Since all our functions have special handlers and no BODY, NLOC is unused,
+*  so we repurpose it.
+*
+*/
+
+#ifdef SYS_IS_64_BIT
+#define LENSTART_WIDTH 8
+#define WIDTH_WIDTH 6
+#define MAX_FIELDS 10
+#else
+#define LENSTART_WIDTH 8
+#define WIDTH_WIDTH 5
+#define MAX_FIELDS 6
```

Comment:
Why are there at most 6 fields? I have not yet tried to understand the code after this, but I just looked at it, and it seems quite complicated... But why is that necessary? In my imagination, it would be sufficient to do this: For each field, the getter/setter need exactly two things: the offset of the field, and a mask (or alternatively, the number of bits in the fields, but I think the mask is more efficient).

The mask takes up one Obj, the offset another (but the offset fits in a single byte, so it might be easy to stash it somewhere else). If we are really concerned about space usage, we can store an offset and field with in 6 bit each.

Assuming we stored the mask and offset, the generic setter andgetter function would look like this (in untested pseudo code, but I hope it's enough to get the basic idea across); note that this does not use INTOBJ_INT or INT_INTOBJ at, because if we setup the mask and offset suitably, they are not necessary.
```C
Obj SetterFunc(Obj self, Obj packedValue, Obj fieldValue)
{
   GAP_ASSERT(IS_INTOBJ(packedValue));
   GAP_ASSERT(IS_INTOBJ(fieldValue));
   UInt mask = extractMaskFromFunc(self);
   UInt offset = extractOffsetFromFunc(self);
   UInt v = ((UInt)fieldValue ^ 1) << offset;
   UInt newVal = ((((UInt)packedValue) & ~mask) | v;
   return (Obj)newVal;
}
Obj GetterFunc(Obj self, Obj packedValue)
{
   GAP_ASSERT(IS_INTOBJ(packedValue));
   UInt mask = extractMaskFromFunc(self);
   UInt offset = extractOffsetFromFunc(self);
   UInt newVal = ((((UInt)packedValue) & mask) >> offset) | 1;
   return (Obj)newVal;
}
```